src/main/native/lib/reader/impl/remote_block_reader.h

```diff
@@ -195,8 +204,26 @@ struct RemoteBlockReader<Stream>::ReadData : monad::Monad<> {
       }
       next(status);
     };
-    auto data_len = self_->header_.datalen();
-    async_read(*self_->stream_, buf_, asio::transfer_exactly(data_len), handler);
+
+    auto readdata = [next, this, handler](const asio::error_code &ec, size_t transferred) {
+      //padding has been consumed, can read data directly into buffer
```

Comment:
I need to add a check on the asio error code and an assertion that transferred is the number of padding bytes here.  Accidentally got rid of that when I stripped out debugging stuff.